src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -258,3 +264,117 @@ load_events_from_file(FILE* fptr)
     return parsed_events;
 
 }
+
+
+int dump_file_info(PyObject* file_info, FILE* fptr){
+    const char * buffer = NULL;
+    uint64_t content_length = 0;
+    uint64_t compression_length = 0;
+    PyObject* json_ret = NULL;
+    PyObject* zlib_ret = NULL;
+    PyObject* bytes_data = NULL;
+    PyObject* dumps_func = PyObject_GetAttrString(json_module, "dumps");
+    PyObject* json_args = PyTuple_New(1);
+    PyObject* compress_func = PyObject_GetAttrString(zlib_module, "compress");
+    PyObject* zlib_args = PyTuple_New(1);
+
+    if (!dumps_func) {
+        perror("Failed to access json.dumps()");
+        exit(-1);
+    }
+
+    if (!compress_func) {
+        perror("Failed to access zlib.compress()");
+        exit(-1);
+    }
+    
+    // json dumps file_info
+    PyTuple_SetItem(json_args, 0, file_info);
+    Py_INCREF(file_info); 
+    json_ret = PyObject_CallObject(dumps_func, json_args);
+
+    // convert string to bytes
+    bytes_data = PyObject_CallMethod(json_ret, "encode", NULL);
+    content_length = PyBytes_Size(bytes_data);
+
+    // compress bytes data
+    PyTuple_SetItem(zlib_args, 0, bytes_data);
+    zlib_ret = PyObject_CallObject(compress_func, zlib_args);
+    compression_length = PyBytes_Size(zlib_ret);
+    buffer = PyBytes_AsString(zlib_ret);
+
+    // write data
+    fputc(VC_HEADER_FILE_INFO, fptr);
+    fwrite(&compression_length, sizeof(uint64_t), 1, fptr);
+    fwrite(&content_length, sizeof(uint64_t), 1, fptr);
+    fwrite(buffer, sizeof(char), compression_length, fptr);
+
+    Py_DECREF(zlib_ret);
```

Comment:
It's better to release the resource as soon as it's no longer in use. For example, `bytes_data`'s referece is stolen by `PyTuple_SetItem(zlib_args, 0, bytes_data)`, so releasing `zlib_args` should release `bytes_data` as well. Releasing everyting at end is making it very difficult to verify the logic.